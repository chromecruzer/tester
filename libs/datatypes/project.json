{
  "name": "datatypes",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/datatypes/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/datatypes/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/datatypes"],
      "options": {
        "jestConfig": "libs/datatypes/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
