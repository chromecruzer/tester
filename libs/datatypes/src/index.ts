export * from './datatypes';
export * from './IolRecord';
export * from './CustomerRecord';
export * from './CustomerContactRecord';
export * from './ProductRecord';
export * from './EmployeeRecord';        // added by shyam for testing 
export * from './EmployeeRecordDataMappings'   //
export * from './UserRecord1'  //
export * from './UserRecordDM' // 
export * from './ConsignmentRecord';
export * from './CustomerDataMapping';
export * from './CustomerContactRecord';
export * from './CustomerContactDataMapping';
export * from './IolDataMapping';
export * from './notifications';
export * from './ProductDataMapping';
export * from './SalesRepDataMapping';
export * from './SalesRepRecord';
export * from './SalesMappingSquared';
export * from './SalesMappingRecord';
export * from './SalesJumpRecord';
export * from './UploadHeaders';
export * from './AuditRecord';
export * from './SearchSignature';
export * from './EmailSettings';
//export * from './UserRecord';
import BuildUrl from "./BuildUrl";
import {ExpirationCalculations} from "./ExpirationCalculations";
import ItemMatchStatus from './ItemMatchStatus';
import SearchSignature from "./SearchSignature";
export {BuildUrl, ExpirationCalculations, ItemMatchStatus, SearchSignature};
