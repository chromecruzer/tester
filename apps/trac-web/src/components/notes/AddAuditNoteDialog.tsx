import {useState} from "react";
import {DialogModal} from "../tables/DialogModal";
import {dateNow, formatTracDate, NullableString} from "@trac/datatypes";
import {useMutation, useQueryClient} from "@tanstack/react-query";
import {getAudit, postAuditNotes} from "../../api/auditApi";

export const AddAuditNoteDialog = ({author, location, open, requestClose, auditUuid}) => {
  const [note, setNote] = useState('');
  const [error, setError] = useState(null as NullableString);
  const [isLoading, setIsLoading] = useState(false);
  const queryClient = useQueryClient();
  const postAuditNotesMutation = useMutation({
    mutationFn: async () => {
      setIsLoading(true);
      setError(null);
      handleClose();
      return postAuditNotes([{
        date_created: formatTracDate(dateNow()),
        author,
        annotate_type: 'Audit',
        annotated_uuid: location.uuid,
        audit_uuid: location.uuid,
        content: note,
      }]);
    },
    onSuccess: (data, variables, context) => {
      setError(null);
      queryClient.prefetchQuery(['audit'], () => getAudit(auditUuid));
    },
    onError: (error:NullableString, variables, context) => {
      if(error) {
        setError(error);
        console.error('post audit notes',error);
      }
    },
    onSettled: (data, error, variables, context) => {
      setIsLoading(false);
    }
  })
  const handleClose = () => {
    setNote('');
    requestClose();
  }
  const handleChange = e => {
    setNote(e.target.value);
  }
  const handleSubmit = () => {
    postAuditNotesMutation.mutate();
  }
  return <DialogModal open={open} requestClose={handleClose}>
    <div className='flex-col min-h-full min-w-full justify-items-stretch'>
      <h2 className='px-2 py-2 text-center font-bold text-bl-text-dark'>Add A Note To The Audit</h2>
      <textarea
        value={note}
        rows={12}
        placeholder='Type your note here'
        onChange={handleChange}
        className='border-2 px-2 py-2 grow justify-self-stretch self-stretch min-h-0 min-w-0 w-full h-full'
      />
      <div className='flex-row content-between'>
        <button className='border-2 px-2 py-2 filter-button-active' onClick={() => handleClose()}>Cancel</button>
        <button className='border-2 px-2 py-2 filter-button-active' onClick={handleSubmit}>Ok</button>
      </div>
    </div>
  </DialogModal>
}
