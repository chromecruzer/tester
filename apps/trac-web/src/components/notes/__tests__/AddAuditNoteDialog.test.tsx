import * as MockHooks from "../../../redux/hooks";
import * as MockAuditActions from "../../../redux/auditActions";
import userEvent from "@testing-library/user-event";
import {render, screen} from "@testing-library/react";
import {useState as MockUseState} from "react";
import {AddAuditNoteDialog} from "../AddAuditNoteDialog";
import {DateTime} from "luxon";
import {tracDateFormat} from "@trac/datatypes";

jest.mock('react', () => ({
  ... jest.requireActual('react'),
  useState: jest.fn()
}))
jest.mock('../../tables/DialogModal', () => ({
  DialogModal: ({open, requestClose, children}) => (
    <div>
      <h4>open is set to {open}</h4>
      <h4>requestClose is {requestClose ? 'present' : 'absent'}</h4>
      {children}
    </div>)
}))
describe('AddAuditNoteDialog', () => {
  let closeSpy, dispatchSpy, postAuditNotesSpy, setNoteSpy;
  const renderAddAuditNoteDialog = () => {
    render(<AddAuditNoteDialog open={true} requestClose={closeSpy} author={'Beethoven'} location={{uuid:'Vienna'}}/>)
  }
  const useStateReturns = (content) => {
    (MockUseState as jest.Mock).mockImplementation(() => [content, setNoteSpy]);

  }
  beforeEach(() => {
    closeSpy = jest.fn();
    dispatchSpy = jest.fn();
    postAuditNotesSpy = jest.fn();
    setNoteSpy = jest.fn();
    jest.spyOn(MockHooks, 'useAppDispatch').mockImplementation(() => dispatchSpy);
    postAuditNotesSpy = jest.spyOn(MockAuditActions, 'postAuditNotes').mockReturnValue(() => Promise.resolve());
    jest.spyOn(DateTime, 'now').mockImplementation(() => DateTime.fromFormat('07/20/2022',tracDateFormat))
  });
  it('should save a typed note', () => {
    const expected = 'Adding an overall note\n Auf Wiedersehn';
    userEvent.setup();
    useStateReturns(expected);
    renderAddAuditNoteDialog();
    const textArea = screen.getByRole('textbox');
    return userEvent.type(textArea, '!')
      .then(() => {
        expect(setNoteSpy).toBeCalledWith(`${expected}!`);
      })
  });
  it('should submit a typed note', () => {
    const expected = 'Ich bin fertig mit meinen note';
    userEvent.setup();
    useStateReturns(expected);
    renderAddAuditNoteDialog();
    const button = screen.getByRole('button', {name: 'Ok'});
    return userEvent.click(button)
      .then(() => {
        expect(postAuditNotesSpy).toBeCalled();
        expect(dispatchSpy).toBeCalled();
        expect(closeSpy).toBeCalled();
        expect(postAuditNotesSpy.mock.calls).toMatchSnapshot();
      });
  });
});
