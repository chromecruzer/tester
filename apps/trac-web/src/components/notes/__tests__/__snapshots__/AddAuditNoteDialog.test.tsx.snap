// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAuditNoteDialog should submit a typed note 1`] = `
Array [
  Array [
    Array [
      Object {
        "annotate_type": "Audit",
        "annotated_uuid": "Vienna",
        "audit_uuid": "Vienna",
        "author": "Beethoven",
        "content": "Ich bin fertig mit meinen note",
        "date_created": "07/20/2022",
      },
    ],
  ],
]
`;
